File: src/main/kotlin/onl/tesseract/srp/skill/domain/model/skill/Skill.java
package onl.tesseract.srp.skill.domain.model.skill;

import onl.tesseract.srp.skill.domain.model.recipe.Tier;
import org.jetbrains.annotations.NotNull;

import java.util.Map;

public record Skill(
        SkillName name,
        SkillStructureName structure,
        @NotNull
        Map<@NotNull Tier,@NotNull SkillTier> tiers
) {
}

File: src/main/kotlin/onl/tesseract/srp/domain/skill/Skill.kt
package onl.tesseract.srp.domain.skill

import onl.tesseract.srp.domain.skill.recipe.Recipe

data class Skill(
    val recipe : Map<Int,SkillTier>,
    val structureName: String,
    val name: String
)

data class SkillTier(
    val recipes: Map<Int, Recipe>,
    val recipeByName: Map<String, Recipe>
)